assets/js/googlesitekit/datastore/forms/index.test.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * core/forms data store tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createTestRegistry } from 'tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/forms store', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setValues', () => {
+			it( 'requires the formName param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues();
+				} ).toThrow( 'formName is required for setting values.' );
+			} );
+
+			it( 'requires the formData param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name' );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'requires the formData param to be an object not an array', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name', [] );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'requires the formData param to be an object not a string', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'form-name', 'formData' );
+				} ).toThrow( 'formData must be an object.' );
+			} );
+
+			it( 'works with formData where the key value is updated, and the untouched key and updated key return expected values', () => {
+				const formData = { key1: 'value1', key2: 'value2' };
+				const newFormData = { key1: 'value3' };
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );
+
+				registry.dispatch( STORE_NAME ).setValues( 'form-name', newFormData );
+
+				const formOriginalValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key2' );
+				expect( formOriginalValue ).toEqual( 'value2' );
+
+				const formUpdatedValue = registry.select( STORE_NAME ).getValue( 'form-name', 'key1' );
+				expect( formUpdatedValue ).toEqual( 'value3' );
+			} );
+		} );
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getValue', () => {
+			it( 'works with a formName that does not exist', () => {
```

Comment:
This test name implies that `getValue( 'form-name' )` works even if `registry.dispatch( STORE_NAME ).setValues( 'form-name', {} )` has never been called, but before you check the result of `getValue` you _do_ set data for a different `formName`. Because they're so closely named and the setting of the form name doesn't impact the test at all, this made me do a double-take.

At first, I thought this test wasn't testing the appropriate behaviour, but then I saw the differing `formName` values and saw that it is. But why is `registry.dispatch( STORE_NAME ).setValues( 'form-name', formData );` being called at all? It doesn't impact this test but makes it confusing to read.

This test can be simplified and made less confusing to parse 🙂, eg:

```js
it( 'works with a formName that does not exist', () => {
  const formValue = registry.select( STORE_NAME ).getValue( 'new-form-name', 'key2' );
  expect( formValue ).toEqual( undefined );
} );
```